client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -214,6 +213,7 @@ public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
                 log.warn("Encountered exception invoking ReplyProcessor for flow id {}", flowId, e);
             }
         });
+        registeredFutureLatch.releaseExceptionally(new ConnectionClosedException());
```

Comment:
https://github.com/pravega/pravega/blob/177eed2f187d7b88a7084619d3ad5bef162ff282/client/src/main/java/io/pravega/client/segment/impl/AsyncSegmentInputStreamImpl.java#L171-L178

This would prevent segment readers from retrying, right?

Reply:
That getConnection call on line 180 is going to end up causing the retry to happen on a new connection. This one connection won't be re-used once it is unregistered, the client will simply create a new one. I believe we don't currently have any code paths that attempt to re-use an unregistered connection. Please let me know if I am wrong about that.

Reply:
Took a second look at the code flow, yes we do not have any code path that attempts to re-use an unregistered connection.